Unsuppressible mypy syntax error for azure.core when using it from a typechecked project with python 3.9.0

- **Package Name**: azure.core
- **Package Version**: 1.9.0
- **Operating System**: Arch Linux
- **Python Version**: 3.9.0

**Describe the bug**
Importing from `azure.core.*` will cause a hard-to-suppress mypy typechecking error about internal type annotations of azure core that have syntax errors.

I don't understand why it only happens on 3.9.0.

Note that mypy has a really hard time with syntax errors: https://github.com/python/mypy/issues/6897 which makes it hard for the end user to ignore them.

**To Reproduce**
1. Clone https://github.com/EncircleInc/azure-core-mypy-error
2.  `pip install -r requirements.lock.txt` file.
3. `mypy -p test`. You'll get something like `.../python3.9/site-packages/azure/core/pipeline/policies/_authentication.py:33: error: syntax error in type comment '(TokenCredential, *str, Mapping[str, Any]) -> None`
4. Try to suppress the error by adding `ignore_errors = True` in the `mypy.ini` file.
5. `mypy -p test` still errors out.

**Expected behavior**
No type errors when using default mypy settings, or an error saying there aren't any types available.

--- a/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py
+++ b/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py
@@ -31,7 +31,7 @@ class _BearerTokenCredentialPolicyBase(object):
     """
 
     def __init__(self, credential, *scopes, **kwargs):  # pylint:disable=unused-argument
-        # type: (TokenCredential, *str, Mapping[str, Any]) -> None
+        # type: (TokenCredential, *str, **Any) -> None
         super(_BearerTokenCredentialPolicyBase, self).__init__()
         self._scopes = scopes
         self._credential = credential
@@ -103,7 +103,7 @@ class AzureKeyCredentialPolicy(SansIOHTTPPolicy):
     :raises: ValueError or TypeError
     """
     def __init__(self, credential, name, **kwargs):  # pylint: disable=unused-argument
-        # type: (AzureKeyCredential, str, Any) -> None
+        # type: (AzureKeyCredential, str, **Any) -> None
         super(AzureKeyCredentialPolicy, self).__init__()
         self._credential = credential
         if not name:

--- a/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py
+++ b/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py
@@ -16,7 +16,7 @@ except ImportError:
 
 if TYPE_CHECKING:
     # pylint:disable=unused-import
-    from typing import Any, Dict, Mapping, Optional
+    from typing import Any, Dict, Optional
     from azure.core.credentials import AccessToken, TokenCredential, AzureKeyCredential
     from azure.core.pipeline import PipelineRequest
 

